src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -307,50 +296,49 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, turbconv_model)
                     (entr_detrʲ_prev_level.entr - entr_detrʲ_prev_level.detr)
                 )
 
-            @. ρaʲu³ʲ_datau³ʲ_data =
+            @. u³ʲ_datau³ʲ_data =
                 (1 / local_geometry_halflevel.J^2) * (
                     local_geometry_prev_halflevel.J^2 *
-                    ρaʲ_prev_level *
                     u³ʲ_data_prev_halflevel *
                     u³ʲ_data_prev_halflevel
                 )
 
-            @. ρaʲu³ʲ_datau³ʲ_data -=
+            @. u³ʲ_datau³ʲ_data -=
                 (1 / local_geometry_halflevel.J^2) * (
                     local_geometry_prev_level.J^2 *
-                    ρaʲ_prev_level *
-                    ((
-                        ∇p_perturb³_prev_level_data / ρʲ_prev_level +
-                        ∇Φ³_prev_level_data *
-                        (ρʲ_prev_level - ρ_ref_prev_level) / ρʲ_prev_level
-                    ))
+                    FT(2) *
+                    (
+                        ∇Φ³_prev_level_data * (ρʲ_prev_level - ρ_prev_level) /
+                        ρ_prev_level
+                    )
                 )
 
-            @. ρaʲu³ʲ_datau³ʲ_data +=
+            @. u³ʲ_datau³ʲ_data +=
                 (1 / local_geometry_halflevel.J^2) * (
                     local_geometry_prev_level.J^2 *
-                    ρaʲ_prev_level *
+                    FT(2) *
                     (
                         entr_detrʲ_prev_level.entr * u³⁰_data_prev_halflevel -
-                        entr_detrʲ_prev_level.detr * u³ʲ_data_prev_halflevel
+                        entr_detrʲ_prev_level.entr * u³ʲ_data_prev_halflevel
                     )
                 )
 
-            @. ρaʲ_level = ifelse(
+            @. u³ʲ_halflevel = ifelse(
                 (
-                    ρaʲu³ʲ_datau³ʲ_data < (FT(1e-6) / ∂x³∂ξ³_level^2) ||
+                    u³ʲ_datau³ʲ_data < (FT(1e-6) / ∂x³∂ξ³_level^2) ||
```

Comment:
Do you want to keep the same threshold for `ρaʲu³ʲ_data` and `u³ʲ_datau³ʲ_data`?

Reply:
Yeah I basically made up this threshold because I don't know what would be a good number. Right now it corresponds to w^2 < 1e-6, so w < 1e-3, where w is the physical velocity. I guess it sounds reasonable?